packages/remix-dev/compiler.ts

```diff
@@ -417,15 +421,11 @@ async function createServerBuild(
       entryPoints,
       outfile: config.serverBuildPath,
       write: false,
+      conditions: isCloudflareWorkersOrPages ? ['worker'] : [],
```

Comment:
I believe we need the other "standard" fields listed as well, along with the default being "undefined" otherwise they won't ever be resolved.

Reply:
Thanks for reviewing 👍🏼  I marked the default to be `undefined` now. But I am not sure what the "standard" fields to be put here.

On esbuild documentation about [conditions](https://esbuild.github.io/api/#how-conditions-work), it mentions `there are five conditions with special behavior that are built in to esbuild, and cannot be disabled`. While on the [commit](https://github.com/evanw/esbuild/commit/1a458e0bf581737e4962e8d33f0af01672b708e4#diff-ea8af26844b5be484f625fbe35bfe3be845e32d4a87d01b8dd9f8b3f3b3e6a37R740) that implement custom conditions for "exports" seems to be doing an append.

Do you think if `['worker']` could work without any standard fields?

Reply:
Yep, looks like are good to go with this. I gave it a run-through locally but the part I don't like is that it's based on the order of the published package. This means that if the package.json for the module looks like this:
```json
{
  "name": "fake-module",
  "version": "1.0.0",
  "main": "./require.js",
  "module": "./module.js",
  "exports": {
    "require": "./require.js",
    "import": "./import.js",
    "worker": "./worker.js",
    "default": "./default.js"
  }
}
```
it will still resolve the "import" version as it's based on the order of the entries in the "exports" object. (this is just how the ecosystem is though 😮‍💨)
It's up the the package author to make sure their exports look more like:
```json
{
  "name": "fake-module",
  "version": "1.0.0",
  "main": "./require.js",
  "module": "./module.js",
  "exports": {
    "worker": "./worker.js",
    "require": "./require.js",
    "import": "./import.js",
    "default": "./default.js"
  }
}
```